src/pages/Bets/components/BetModal.tsx

```diff
@@ -34,25 +35,78 @@ const useSettleContractWrite = ({
   marketAddress,
   index
 }: useSettleContractWriteArgs) => {
-  const { data, error, write } = useContractWrite({
+  const {
+    data,
+    error: settleBetError,
+    write: settleBetWrite
+  } = useContractWrite({
     mode: "recklesslyUnprepared",
     address: marketAddress || "",
     abi: marketContractJson.abi,
     functionName: "settle",
     args: [index]
   });
 
-  const txHash = data?.hash;
-  const { isLoading: isTxLoading, isSuccess: isTxSuccess } =
+  const settleBetTxHash = data?.hash;
+  const { isLoading: isSettleBetTxLoading, isSuccess: isSettleBetTxSuccess } =
     useWaitForTransaction({
-      hash: txHash
+      hash: settleBetTxHash
     });
   return {
-    write,
-    error,
-    isTxLoading,
-    isTxSuccess,
-    txHash
+    settleBetWrite,
+    settleBetError,
+    isSettleBetTxLoading,
+    isSettleBetTxSuccess,
+    settleBetTxHash
+  };
+};
+
+type useMarketOracleResultWriteArgs = {
+  marketAddress?: string;
+  market_id?: string;
+  winningPropositionId?: string;
+  signature?: EcSignature;
+};
+const useMarketOracleResultWrite = ({
+  market_id,
+  winningPropositionId,
+  signature
+}: useMarketOracleResultWriteArgs) => {
+  const {
+    data,
+    error: marketOracleError,
+    write: setResultMarketOracleWrite
+  } = useContractWrite({
+    mode: "recklesslyUnprepared",
+    addressOrName:
+      process.env.REACT_APP_MARKET_ORACLE_CONTRACT ||
+      "0x592a44ebad029EBFff3Ee4950f1E74538a19a2ea",
+    contractInterface: marketOracleContractJson.abi,
```

Comment:
on development branch, we've upgrade wagmi to v0.8
need to change params according to new wagmi hook signature
addressOrName -> address
contractInterface -> abi